logstash-core/lib/logstash/state_resolver.rb

```diff
@@ -41,14 +41,19 @@ def resolve(pipelines_registry, pipeline_configs)
         end
       end
 
-      configured_pipelines = pipeline_configs.map { |config| config.pipeline_id.to_sym }
+      configured_pipelines = pipeline_configs.map { |config| config.pipeline_id.to_sym }.to_set
```

Comment:
I am assuming `to_set` here is to dedup values but under which condition are there duplicates in this collection??

Reply:
the `set` is for O(1) `configured_pipelines.include?`. `configured_pipelines` was a Array

Reply:
Right, ok. But TBH given the small number of pipelines we are dealing with this extra `to_set`  conversion is practically & *possibly* more costly than the actual `include?` iterations.

To push that micro optimization further I would try to avoid the Array -> Array -> Set and directly produce a `Hash` or `Set` using `inject` or `each_with_object` ?

Something like?
```ruby
configured_pipelines = pipeline_configs.each_with_object(Set.new) { |config, set| set.add(config.pipeline_id.to_sym) }
```